modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/Sequence.scala

```diff
@@ -57,27 +59,43 @@ object Sequence {
     val next: Option[Zipper] =
       focus.next match {
         // Step completed
-        case None      =>
-          pending match {
+        case None      => {
+          remaining match {
             case Nil             => None
```

Comment:
The caller to `next` will produce a `Final` state when the result is `None`. The construction of `Final` takes care of marking the last steps as skipped.